app/main.py

```
@@ -1,3 +1,76 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self):
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self):
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other):
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        return NotImplemented
+
+    def __iadd__(self, other):
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        return self
+
+    def __mul__(self, factor):
+        if isinstance(factor, (int, float)):
+            return Distance(self.km * factor)
+        return NotImplemented
+
+    def __truediv__(self, divisor):
+        if isinstance(divisor, (int, float)) and divisor != 0:
```

The condition `divisor != 0` is correct, but it might be beneficial to raise a `ZeroDivisionError` instead of returning `NotImplemented` when the divisor is zero. This would provide a clearer indication of the error to the user.